src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -3533,7 +3533,7 @@ private TdsOperationStatus TryNextResult(out bool more)
                             if (result != TdsOperationStatus.Done)
                             {
                                 more = false;
-                                return TdsOperationStatus.Done;
+                                return result;
```

Comment:
Should this be fixed in main with a separete PR?

Reply:
This originally came from my [Wraith2#5](https://github.com/Wraith2/SqlClient/pull/5) PR. I didn't spot it on the original OperationStatus PR, so decided to submit it for rollup into this one. I'm happy to push it into main directly too though.

Reply:
I'm not sure if it has any implications if left in main

Reply:
Nothing in the current codebase needs to be able to tell the difference between Done and NeedMoreData. That will come in part3 of this series of PRs. So this can be here, or in main as a separate PR, doesn't really matter. It just seemed easiest to incorporate it into this change when @edwardneal pr'ed it into my repo.